src/services/token/readOnlyTokenHook.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ *  TODO: Extract this code out to its own package to reuse in tradetrust-website
+ */
+
+import { ReadOnlyToken } from "@govtechsg/oa-token";
+
+import { getLogger } from "../../utils/logger";
+
+import { useState, useEffect, useCallback } from "react";
+import { WrappedDocument } from "@govtechsg/open-attestation";
+import { useEthereumTransactionState, TransactionState } from "./helpers";
+const { trace, error } = getLogger("useToken");
+
+type UseReadOnlyToken = { state: TransactionState; tokenInstance: ReadOnlyToken | null; getTokenOwner: GetTokenOwner };
+type GetTokenOwner = () => Promise<string>;
+
+export const useReadOnlyToken = ({ document }: { document: WrappedDocument<any> }): UseReadOnlyToken => {
+  trace(`document to initialize ${JSON.stringify(document)}`);
+
+  const [tokenInstance, setTokenInstance] = useState<ReadOnlyToken | null>(null);
+  const { state, setReady, setLoading, setError, setSuccess } = useEthereumTransactionState();
+
+  const setErrorCallback = useCallback(setError, []);
```

Comment:
did you try to use useCallback in the hook instead ? Here it requires every consumer of the hook to create a new function which is memoized ? It's very incovenient